src/java/org/apache/cassandra/tools/SSTableMetadataViewer.java

```diff
@@ -321,6 +323,25 @@ private void printSStableMetadata(File file, boolean scan) throws IOException
         CompactionMetadata compaction = statsComponent.compactionMetadata();
         SerializationHeader.Component header = statsComponent.serializationHeader();
         Class<? extends ICompressor> compressorClass = null;
+
+        TableMetadata metadata = Util.metadataFromSSTable(descriptor);
+        SSTableReader sstable = SSTableReader.openNoValidation(null, descriptor, TableMetadataRef.forOfflineTools(metadata));
+        int count = 0;
+        
+        try(KeyIterator iter = sstable.keyIterator())
+        {
+            while (iter.hasNext()) 
+            {
+                iter.next();
+                count += 1;
+            }
+            iter.close();
+        }
+
+        TimeUnit tsUnit = TimeUnit.MICROSECONDS;
+        long seconds  = stats.maxTimestamp - stats.minTimestamp;
```

Comment:
This is not safe because if you look on line 360, there is `if (stats != null)` so in theory, I guess, it can be null, but here you are already reading timestamps from `stats`.